mesonbuild/backend/backends.py

```diff
@@ -832,15 +832,16 @@ def determine_rpath_dirs(self, target: T.Union[build.BuildTarget, build.CustomTa
     @staticmethod
     @lru_cache(maxsize=None)
     def canonicalize_filename(fname: str) -> str:
-        parts = Path(fname).parts
+        fname = fname.replace('\\', '/')
+        parts = fname.split('/')
```

Comment:
Why not use `os.path.split()`?

Reply:
Because `str.split` splits at every occurrence of the separator, while `os.path.split` only separates the last part of the path

Reply:
Right. In that case I think a better solution is:
```python
if os.altsep is not None:
    fname.replace(os.altsep, os.sep)
parts = fname.split(os.sep)
...
```